server/src/main/java/org/elasticsearch/search/vectors/KnnScoreDocQuery.java

```diff
@@ -72,6 +72,12 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
             throw new IllegalStateException("This KnnScoreDocQuery was created by a different reader");
         }
         return new Weight(this) {
+
+            @Override
+            public int count(LeafReaderContext context) {
+                return segmentStarts[context.ord + 1] - segmentStarts[context.ord];
+            }
```

Comment:
does the test below still succeed if you remove this count implementation? This is maybe tricky to test, I wonder if we are testing that we override count.

Reply:
IMO, we simply need to make sure that `count` is the same as the actual iterator count. Which is what the test is doing. Of course, the test passed before this.